src/CMakeLists.txt

```diff
@@ -548,9 +548,12 @@ if(BUILD_CLIENT)
 		${PLATFORM_LIBS}
 		${CLIENT_PLATFORM_LIBS}
 	)
-	target_link_libraries(
-		${client_LIBS}
+	set_target_properties(${PROJECT_NAME} PROPERTIES
+		# This is necessary for dynamic Lua modules
+		# to work when Lua is statically linked (issue #10806)
+		ENABLE_EXPORTS 1
```

Comment:
I guess this is because Lua is embedded and built into Minetest?

What's the disadvantage of exporting all of Minetest like this?

Could you do this better by making the embedded Lua into a CMake target which Minetest depends on, and then set ENABLE_EXPORTS 1 just for that target

Reply:
Tried that first, doesn't work.

> I guess this is because Lua is embedded and built into Minetest?

Yes. Personally I'd declare such a configuration as non supported but it did work before but :shrug:.
Alternative would be to actually do that and allow Minetest to link to Lua 5.1 as provided by the system.

Reply:
> What's the disadvantage of exporting all of Minetest like this

+1MiB to the executable size as it now contains 10k entries (46k in debug build) in its .dynsym instead of just 1k. It may also affect startup time but I don’t see any difference.

> Could you do this better by making the embedded Lua into a CMake target which Minetest depends on, and then set ENABLE_EXPORTS 1 just for that target

That won’t work, ENABLE_EXPORTS is specifically for executables. What could work is also adding `-fvisibility=hidden` to compile options on all code but Lua itself.

Reply:
Well :+1: then. If it causes issues, I'm sure people will file bug reports

Reply:
FYI before #10745 CMake did this implicitly, so this isn't something *new*.

Reply:
Ah, I suspected that